Common/Thread/ThreadManager.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+
+// The new threadpool.
+
+// To help future smart scheduling.
+enum class TaskType {
+	CPU_COMPUTE,
+	IO_BLOCKING,
+};
+
+// Implement this to make something that you can run on the thread manager.
+class Task {
+public:
+	virtual ~Task() {}
+	virtual void Run() = 0;
+	virtual bool Cancellable() { return false; }
+	virtual void Cancel() {}
+	virtual float Priority() { return 1.0f; }
```

Comment:
Nothing in this pull considers Priority() yet right?  With more threads, it is less important...

-[Unknown]

Reply:
No, it doesn't. I'll get rid of it I think, we can put it back when/if we need it.